rfc/attestation-config.md

```diff
@@ -0,0 +1,76 @@
+# Attestation configuration options
+
+To allow users more in-depth control over validating attestation statements, a separate, attestation-specific entry in the user's Constellation config file needs to be added.
+
+This option should allow for more in-depth configuration of how and what parts of an attestation statement are validated.
+For example, one should be able to set the minimum microcode version of an SEV-SNP attestation statement, that is deemed acceptable,
+or configure acceptable TCB statuses for Intel TDX attestation.
+
+## In the user's config
+
+A new `attestation` entry in the config file will be added, holding the options used for attestation validation.
+Existing attestation options, like `measurements` and `idKeyDigests` will be moved to this entry.
+
+This entry will replace the `attestationVariant` entry.
+
+Example configuration for Azure SEV-SNP:
+
+```yaml
+version: v3
+image: vX.Y.Z
+name: constell
+stateDiskSizeGB: 30
+kubernetesVersion: "v1.25.6"
+debugCluster: false
+provider:
+  azure: {} # Configuration for creating Azure cloud resources
+# Configuration for attestation validation. This configuration provides sensible defaults for the Constellation version it was created for.
+# See our docs for an overview on attestation: https://docs.edgeless.systems/constellation/architecture/attestation
+attestation:
+  # Azure SEV-SNP attestation configuration
+  # See our docs for information on about configurable values [Link to our docs explaining SEV-SNP on Azure]
+  azureSEVSNP:
```

Comment:
I am wondering, if we need to implement special attestation schemas for every provider (e.g. SNP on Azure and on GCP cannot be the same due to MAA), wouldn't it make sense to move this to provider?

Reply:
Okay, I don't have strong opinions on this so I am fine either way.